zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/CassandraSpanConsumer.java

```diff
@@ -80,74 +97,150 @@
    * This fans out into many requests, last count was 8 * spans.size. If any of these fail, the
    * returned future will fail. Most callers drop or log the result.
    */
-  @Override
-  public Call<Void> accept(List<Span> rawSpans) {
+  @Override public Call<Void> accept(List<Span> spans) {
+    int spanCount = spans.size();
+    if (spanCount == 0) return Call.create(null);
+
     V2SpanConverter converter = V2SpanConverter.create();
     V1ThriftSpanWriter encoder = new V1ThriftSpanWriter();
 
     Set<InsertTrace.Input> insertTraces = new LinkedHashSet<>();
-    Set<String> insertServiceNames = new LinkedHashSet<>();
-    Set<InsertRemoteServiceName.Input> insertRemoteServiceNames = new LinkedHashSet<>();
-    Set<InsertSpanName.Input> insertSpanNames = new LinkedHashSet<>();
-    Set<Map.Entry<String, String>> autocompleteTags = new LinkedHashSet<>();
 
     List<Call<Void>> calls = new ArrayList<>();
-    for (Span v2 : rawSpans) {
-      V1Span span = converter.convert(v2);
-      // indexing occurs by timestamp, so derive one if not present.
-      long ts_micro = v2.timestampAsLong();
-      if (ts_micro == 0L) ts_micro = guessTimestamp(v2);
+    for (int i = 0; i < spanCount; i++) {
+      Span span = spans.get(i);
+      V1Span v1Span = converter.convert(span);
 
-      insertTraces.add(insertTrace.newInput(span, ByteBuffer.wrap(encoder.write(v2)), ts_micro));
+      // trace records need an insertion timestamp, so derive one if not present.
+      long ts_micro = span.timestampAsLong();
```

Comment:
This is copied from before, but seems to be weird naming convention (unless it's referring to a virus scan)

Reply:
:) these are the field names in the cassandra schema